components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt

```diff
@@ -85,6 +86,22 @@ class CrashReporter(
         logger.info("Crash report submitted to ${services.size} services")
     }
 
+    /**
+     * Add a crash breadcrumb to all registered services with breadcrumb support.
+     *
+     * ```Kotlin
+     *   crashReporter.recordCrashBreadcrumb(
+     *       CrashBreadcrumb("Settings button clicked", data, "UI", Level.INFO, Type.USER)
+     *   )
+     * ```
+     */
+    fun recordCrashBreadcrumb(crashBreadcrumb: CrashBreadcrumb) {
+        if (crashBreadcrumbList.size >= BREADCRUMB_MAX_NUM) {
+            crashBreadcrumbList.removeAt(0)
+        }
+        crashBreadcrumbList.add(crashBreadcrumb)
+    }
```

Comment:
Agreed, priority could be a good way of deciding what to drop.  I think we will need more discussions though.  This could be something that's configurable.  I'll raise a issue to track this.

Reply:
https://github.com/mozilla-mobile/android-components/issues/4299